src/test/javascript/spec/component/shared/result-detail.component.spec.ts

```diff
@@ -163,24 +177,49 @@ describe('ResultDetailComponent', () => {
                     type: ExerciseType.PROGRAMMING,
                     staticCodeAnalysisEnabled: true,
                     maxStaticCodeAnalysisPenalty: 20,
+                    projectKey: 'somekey',
                 } as ProgrammingExercise;
 
                 comp.result = {
                     id: 89,
                     participation: {
                         id: 55,
                         exercise,
+                        type: ParticipationType.PROGRAMMING,
+                        participantIdentifier: 'student42',
+                        repositoryUrl: 'https://bitbucket.ase.in.tum.de/projects/somekey/repos/somekey-student42',
                     },
                 } as Result;
 
                 buildLogService = debugElement.injector.get(BuildLogService);
                 resultService = debugElement.injector.get(ResultService);
+                profileService = debugElement.injector.get(ProfileService);
 
                 buildlogsStub = stub(buildLogService, 'getBuildLogs').returns(of([]));
                 getFeedbackDetailsForResultStub = stub(resultService, 'getFeedbackDetailsForResult').returns(of({ body: [] as Feedback[] } as HttpResponse<Feedback[]>));
+
+                // Set profile info
+                const profileInfo = new ProfileInfo();
+                profileInfo.commitHashURLTemplate = commitHashURLTemplate;
+                stub(profileService, 'getProfileInfo').returns(new BehaviorSubject(profileInfo));
             });
     });
 
+    it('should generate commit link for programming exercise result with submission, participation and exercise', () => {
```

Comment:
Could you also add tests for the edge cases (n/a or empty string/undefined)?

Reply:
I can look it up, but the commit link generating function is already tested somewhere else (where it originally came from).

Reply:
But didn't you move the test as well?

Reply:
No, the test is entirely new, to check this component specifically.

Reply:
Took another closer look, now I get it, sorry.

If you move the other one, I think it makes sense to add those two small tests though. Then this function is properly tested.